src/qs8-gemm/neon-mlal-lane.c.in

```diff
@@ -318,39 +347,81 @@ void xnn_${DATATYPE_SPEC}_gemm_minmax${REQUANTIZATION_SPEC}_ukernel_${MR}x${NR}_
       }
     }
 
-    $if DATATYPE in ["QD8", "QC4_F16", "QC4_F32"]:
-      $for M in range(0, MR):
+    $if BLOCKWISE:
+      $for N in range(0, NR, 4):
+        const float32x4_t vfilter_output_scale${ABC[N:N+4]} = vld1q_f32(w); w = (const float*) w + 4;
+
+      $for M in range(MR):
         $for N in range(0, NR, 4):
-          $if DATATYPE in ["QC4_F16", "QC4_F32"]:
-            float32x4_t vout${M}x${ABC[N:N+4]} = vcvtq_n_f32_s32(vacc${M}x${ABC[N:N+4]}, 4);
-          $else:
-            float32x4_t vout${M}x${ABC[N:N+4]} = vcvtq_f32_s32(vacc${M}x${ABC[N:N+4]});
+          const float32x4_t vf${M}x${ABC[N:N+4]} = vcvtq_f32_s32(vacc${M}x${ABC[N:N+4]});
 
-      $for M in range(0, MR, 2):
-        $if M + 1 == MR:
-          const float32x4_t vinput_scale${M} = vld1q_dup_f32(&quantization_params[${M}].inv_scale);
+        $for N in range(0, NR, 4):
```

Comment:
what if you flop the loop in the generator (metakernel), does it help with the register pressure?
i.e.
`
for(MR)
  for(NR)
`
vs.
`
for(NR)
  for(MR)
`  
also you can interleave the scales (which has long life right now) to see if it helps.